tests/e2e/suite_test.go

```diff
@@ -148,6 +162,13 @@ func saveOperatorLogs(buf bytes.Buffer, specName string, output io.Writer) {
 	}
 }
 
+// isUpgradeTestSuite tells if current test suite is an upgrade test suite or not basing on
+// the label filter specified in the Ginkgo CLI with the option "--label-filter"
+func isUpgradeTestSuite() bool {
+	suiteConfig, _ := GinkgoConfiguration()
+	return suiteConfig.LabelFilter == tests.LabelUpgrade
+}
+
```

Comment:
not keen on this. Exceptional flow makes our code more complicated to reason about. I.e. in two months we won't remember that the upgrade run skips client pod creation.
As I understand the issue is with namespace cleanup, let's handle the problem there.
Going to test that hypothesis.